### PR TITLE
Make HostColumnVector.DataType accessor methods public

### PR DESCRIPTION
**Pull request by jbrennan333**
## Description
As part of work on host memory oom handling in the spark-rapids plugin, I am copying (maybe moving) some `ColumnBuilder` code from `HostColumnVector.java` into the plugin.  The following subclasses are referenced by  `ColumnBuilder`: DataType, ListType, StructData, StructType, and BasicType.  These are public static inner classes, but the accessor methods are protected and cannot be accessed from the spark-rapids plugin.  This patch just makes the accessor methods of these classes public.

## Checklist
- [x ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [na] New or existing tests cover these changes.
- [na] The documentation is up to date with these changes.

### CODE REVIEW

File: java/src/main/java/ai/rapids/cudf/HostColumnVector.java
[the remark this answers is not in the document]
Comment (jbrennan333) [reply java/src/main/java/ai/rapids/cudf/HostColumnVector.java:1]: Thanks.  Updated.

**Review by jlowe — APPROVED** [review]

**Review by revans2 — APPROVED** [review]

**Pull request merged**